
From: Nick Piggin <nickpiggin@yahoo.com.au>

Actually turn on SD_LOAD_BALANCE for the regular domains.  Introduced by
5/8 "sched add load balance flag".

Signed-off-by: Nick Piggin <nickpiggin@yahoo.com.au>
Signed-off-by: Andrew Morton <akpm@osdl.org>
---

 25-akpm/include/linux/sched.h |    9 ++++++---
 1 files changed, 6 insertions(+), 3 deletions(-)

diff -puN include/linux/sched.h~sched-enable-sd_load_balance include/linux/sched.h
--- 25/include/linux/sched.h~sched-enable-sd_load_balance	2004-09-26 17:19:15.480183720 -0700
+++ 25-akpm/include/linux/sched.h	2004-09-26 17:19:15.484183112 -0700
@@ -499,7 +499,8 @@ extern void cpu_attach_domain(struct sch
 	.cache_hot_time		= 0,			\
 	.cache_nice_tries	= 0,			\
 	.per_cpu_gain		= 25,			\
-	.flags			= SD_BALANCE_NEWIDLE	\
+	.flags			= SD_LOAD_BALANCE	\
+				| SD_BALANCE_NEWIDLE	\
 				| SD_BALANCE_EXEC	\
 				| SD_WAKE_AFFINE	\
 				| SD_WAKE_IDLE		\
@@ -522,7 +523,8 @@ extern void cpu_attach_domain(struct sch
 	.cache_hot_time		= (5*1000/2),		\
 	.cache_nice_tries	= 1,			\
 	.per_cpu_gain		= 100,			\
-	.flags			= SD_BALANCE_NEWIDLE	\
+	.flags			= SD_LOAD_BALANCE	\
+				| SD_BALANCE_NEWIDLE	\
 				| SD_BALANCE_EXEC	\
 				| SD_WAKE_AFFINE	\
 				| SD_WAKE_BALANCE,	\
@@ -543,7 +545,8 @@ extern void cpu_attach_domain(struct sch
 	.cache_hot_time		= (10*1000),		\
 	.cache_nice_tries	= 1,			\
 	.per_cpu_gain		= 100,			\
-	.flags			= SD_BALANCE_EXEC	\
+	.flags			= SD_LOAD_BALANCE	\
+				| SD_BALANCE_EXEC	\
 				| SD_WAKE_BALANCE,	\
 	.last_balance		= jiffies,		\
 	.balance_interval	= 1,			\
_
